src/SMap.hpp

```diff
@@ -18,17 +18,17 @@ struct SMap
     // Attributes //
     // Bitmaps for each field.
     // If n is a possible digit in field i, then 2^(n-1) & fields[i] is true.
-    uint16_t* fields;
-    bool* blank;
-    bool* fixed;
+    uint16_t fields[9*9];
```

Comment:
Good enough for this PR. Maybe, I will replace those with std::arrays in the future. But this is already an improvement, as I see it.